src/test/kotlin/io/thecontext/ci/validation/EpisodesValidatorSpec.kt

```diff
@@ -0,0 +1,45 @@
+package io.thecontext.ci.validation
+
+import com.greghaskins.spectrum.Spectrum
+import com.greghaskins.spectrum.dsl.specification.Specification.context
+import com.greghaskins.spectrum.dsl.specification.Specification.it
+import io.reactivex.schedulers.Schedulers
+import io.thecontext.ci.memoized
+import io.thecontext.ci.testEpisode
+import org.junit.runner.RunWith
+
+@RunWith(Spectrum::class)
+class EpisodesValidatorSpec {
+    init {
+        val validator by memoized { EpisodesValidator(Schedulers.trampoline()) }
+
+        context("episodes with unique IDs") {
+
+            val episodes = listOf(
+                    testEpisode.copy(id = "id1"),
+                    testEpisode.copy(id = "id2")
+            )
+
+            it("returns validation result successful") {
+                validator.validate(episodes)
+                        .test()
+                        .assertResult(ValidationResult.Success)
+            }
+        }
+
+        context("episodes with duplicating IDs") {
+
+            val episodes = listOf(
+                    testEpisode.copy(id = "id1"),
+                    testEpisode.copy(id = "id2"),
+                    testEpisode.copy(id = "id2")
+            )
+
+            it("returns validation result failure") {
+                validator.validate(episodes)
+                        .test()
+                        .assertValue { it is ValidationResult.Failure }
```

Comment:
Should we check for exact error message as well? Just to be sure that we are handling / catching the expected error an not another Validation Result.Failure

Reply:
Not a fan of copying main code to test code. The issue you are describing should be solved by dedicated `Failure` types, but it is another hell... Let’s agree to use messages in tests everywhere on first false-positive check we’ll encounter as a compromise.